cli/packages/cmd/user.go

```diff
@@ -0,0 +1,248 @@
+	Run:                   func(cmd *cobra.Command, args []string) {},
+}
+
+var switchCmd = &cobra.Command{
+	Use:                   "switch",
+	Short:                 "Used to switch between Infisical profiles",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical switch",
+	Args:                  cobra.ExactArgs(0),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		util.RequireLogin()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		//get previous logged in profiles
+		loggedInProfiles, err := getLoggedInUsers()
+		if err != nil {
+			util.HandleError(err, "[infisical user switch]: Unable to get logged Profiles")
+		}
+
+		//prompt user
+		profile, err := LoggedInUsersPrompt(loggedInProfiles)
+		if err != nil {
+			util.HandleError(err, "[infisical user switch]: Prompt error")
+		}
+
+		//write to config file
+		configFile, err := util.GetConfigFile()
+		if err != nil {
+			util.HandleError(err, "[infisical user switch]: Unable to get config file")
+		}
+
+		configFile.LoggedInUserEmail = profile
+
+		//set logged in user domain
+		ok := util.ConfigContainsEmail(configFile.LoggedInUsers, profile)
+
+		if !ok {
+			//profile not in loggedInUsers
+			configFile.LoggedInUsers = append(configFile.LoggedInUsers, models.LoggedInUser{
+				Email:  profile,
+				Domain: config.INFISICAL_URL,
+			})
+			//set logged in user domain
+			configFile.LoggedInUserDomain = config.INFISICAL_URL
+
+		} else {
+			//exists, set logged in user domain
+			for _, v := range configFile.LoggedInUsers {
+				if profile == v.Email {
+					configFile.LoggedInUserDomain = v.Domain
+					break
+				}
+			}
+		}
+
+		err = util.WriteConfigFile(&configFile)
+		if err != nil {
+			util.HandleError(err, "")
+		}
+	},
+}
+
+var updateCmd = &cobra.Command{
+	Use:                   "update",
+	Short:                 "Used to update properties of an Infisical profile",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical user update",
+	Args:                  cobra.ExactArgs(0),
+	Run:                   func(cmd *cobra.Command, args []string) {},
+}
+
+var domainCmd = &cobra.Command{
+	Use:                   "domain",
+	Short:                 "Used to update the domain of an Infisical profile",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical user update domain",
+	Args:                  cobra.ExactArgs(0),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		util.RequireLogin()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		//prompt for profiles selection
+		loggedInProfiles, err := getLoggedInUsers()
+		if err != nil {
+			util.HandleError(err, "[infisical user update domain]: Unable to get logged Profiles")
+		}
+
+		//prompt user
+		profile, err := LoggedInUsersPrompt(loggedInProfiles)
+		if err != nil {
+			util.HandleError(err, "[infisical user update domain]: Prompt error")
+		}
+
+		domain := ""
+		domainQuery := true
+		if config.INFISICAL_URL_MANUAL_OVERRIDE != util.INFISICAL_DEFAULT_API_URL {
```

Comment:
Can you explain why you choose to use `INFISICAL_URL_MANUAL_OVERRIDE` variable instead of   `config.INFISICAL_URL`? I see in `GetCurrentLoggedInUserDetails`, you set `INFISICAL_URL_MANUAL_OVERRIDE=config.INFISICAL_URL` so i'm curious if we need this or if there is a use case I didn't catch